flink-job/src/main/java/com/ververica/field/dynamicrules/RulesEvaluator.java

```diff
@@ -152,22 +155,31 @@ public void run() throws Exception {
     return RulesSource.Type.valueOf(rulesSource.toUpperCase());
   }
 
-  private StreamExecutionEnvironment configureStreamExecutionEnvironment(
-      RulesSource.Type rulesSourceEnumType, boolean isLocal) {
-    Configuration flinkConfig = new Configuration();
-    flinkConfig.setBoolean(ConfigConstants.LOCAL_START_WEBSERVER, true);
+  private StreamExecutionEnvironment configureStreamExecutionEnvironment() {
+    final String localMode = config.get(LOCAL_EXECUTION);
+
+    StreamExecutionEnvironment env;
+    if (localMode.isEmpty() || localMode.equals(LOCAL_MODE_DISABLE_WEB_UI)) {
+      // cluster mode or disabled web UI
+      env = StreamExecutionEnvironment.getExecutionEnvironment();
+    } else {
+      Configuration flinkConfig = new Configuration();
+      flinkConfig.set(BIND_PORT, localMode);
+      env = StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(flinkConfig);
+    }
 
-    StreamExecutionEnvironment env =
-        isLocal
-            ? StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(flinkConfig)
-            : StreamExecutionEnvironment.getExecutionEnvironment();
+    if (!localMode.isEmpty()) {
```

Comment:
```
--local <port> --rules-source SOCKET
```
If you can live with any port, you can also provide 0 there and it will be printed in the logs (which are currently spammed by printouts though so you cannot find it before that log line is gone; that's something I have as a commit in my performance improvements)

The line you marked here though just contains the restart strategy setting for if `local` is set. The rest is a few lines above and sockets as sources at a completely different place as before.

Reply:
What I mean is you want to have Flink UI in this setup, you cannot set a fixed delay restarts. This is what is needed, because if you have Flink's default restarts. I do not want the flood of Flink messages to overwhelm the IDE if the socket happened to be closed.